eng/pipelines/coreclr/templates/build-perf-maui-apps.yml

```diff
@@ -126,21 +128,21 @@ steps:
 
   - script: |
       chmod -R a+r .
-      ../dotnet.sh publish -bl:MauiAndroid.binlog -f net6.0-android -c Release
+      ../dotnet publish -bl:MauiAndroid.binlog -f net6.0-android -c Release
       mv ./bin/Release/net6.0-android/com.companyname.MauiTesting-Signed.apk ./MauiAndroidDefault.apk
     displayName: Build MAUI Android
     workingDirectory: $(Build.SourcesDirectory)/MauiTesting
 
   - script: |
       chmod -R a+r .
-      ../dotnet.sh publish -bl:MauiiOS.binlog -f net6.0-ios -c Release
+      ../dotnet build -bl:MauiiOS.binlog -f net6.0-ios -c Release
```

Comment:
What I've found to work is:

1. Remove the `net6.0-maccatalyst` TFM from the `TargetFrameworks` property in the .csproj.
2. `dotnet publish -c Release -f net6.0-ios -r ios-arm64`

I've logged https://github.com/xamarin/xamarin-macios/issues/14015 for this, which was closed in favor of https://github.com/dotnet/sdk/issues/21877.